Tests/SwiftDiagnosticsTest/FixItTests.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+final class FixItTests: XCTestCase {
+  func testEditsForFixIt() {
+    let testCases: [[(range: Range<Int>, replacement: String)]] = [
+        [(9..<15, "Multiident "), (15..<21, "")],
```

Comment:
Instead of using hard offsets here, I would prefer to use location markers in the `markedSource`. To get `9`, you can add the location marker `1️⃣` at after `protocol ` and then do 

```swift
let (markers, source) = extractMarkers(markedSource)
let something = markers["1️⃣"] 
```

The advantage of using location markers is that
1. You can easily tell where in the input source code they correspond to without counting characters
2. You don’t need to fix up a bunch of offsets when eg. changing `protocol` to `struct` at some point. It’s just more stable that way.

Reply:
wow, that's a great design. i understand how it works. thanks!